src/core/account/custom-tokens.js

```diff
@@ -247,3 +247,27 @@ export async function saveCustomTokens(
   const file = await customTokensFile.load(disklet, CUSTOM_TOKENS_FILE)
   customTokensFile.save(disklet, CUSTOM_TOKENS_FILE, { ...file, customTokens })
 }
+
+/**
+ * Translates a currency code to a tokenId,
+ * and then back again for bi-directional backwards compatibility.
+ */
+export function upgradeCurrencyCode(
+  ai: ApiInput,
+  accountId: string,
+  walletId: string,
+  info: { currencyCode?: string, tokenId?: string }
+): { currencyCode: string, tokenId?: string } {
+  const walletState = ai.props.state.currency.wallets[walletId]
+  const { currencyInfo, pluginId } = walletState
+  const allTokens = ai.props.state.accounts[accountId].allTokens[pluginId]
+
+  const tokenId =
+    info.tokenId ??
+    Object.keys(allTokens).find(
```

Comment:
Oops. This doesn’t handle when the passed currency code is for the chain itself, like “BTC”